### PR TITLE
feat: support import of Emoji packs as zip file

### PR DESCRIPTION
**Pull request by TheOneWithTheBraid**

### CODE REVIEW

File: lib/pages/settings_emotes/import_archive_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    );
+  }
+
+  _setRenderError() {
+    hasErrorNotifier.value = true;
+    widget.onRemove.call();
+  }
+}
+
+class _ImageFleError extends StatelessWidget {
```
Comment (krille-chan) [inline lib/pages/settings_emotes/import_archive_dialog.dart]: I guess this is a typo and should be "_ImageFileError"?

File: lib/pages/settings_emotes/settings_emotes.dart [hunk 2]
```diff
@@ -262,4 +277,99 @@ class EmotesSettingsController extends State<EmotesSettings> {
...
+            } else {
+              return;
+            }
+          }
+
+          final file = File(savePath);
+          await file.writeAsBytes(output);
+
+          ScaffoldMessenger.of(context).showSnackBar(
+            SnackBar(content: Text('Saved emote pack to $savePath !')),
```
Comment (krille-chan) [inline lib/pages/settings_emotes/settings_emotes.dart]: Missing localizations

File: lib/pages/settings_emotes/import_archive_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+extension on String {
+  String get fileNameOnly {
```
Comment (krille-chan) [inline lib/pages/settings_emotes/import_archive_dialog.dart]: This method is a little bit hard to read and it is not intuitive what it does or why it is needed. Just the filename of a path could just be: `path.split("/").last`

**Review by krille-chan — CHANGES_REQUESTED** [review]

**Comment by TheOneWithTheBraid** [comment]
Fixed the issues you pointed on, can you please check whether you are fine with the changes ?

**Review by krille-chan — APPROVED** [review]

**Pull request merged**